### PR TITLE
Refine Go Language Page

### PR DESCRIPTION
**Pull request by Siddhant-K-code**
Refine the whole Go Language Page

### TODO:

- [x] Add Debugging Section for both IDEs
- [x] Example Project for debugging purpose. Revamp ([old repo](https://github.com/gitpod-io/Gitpod-Go-Debug))

### [**`Live Preview`**](https://deploy-preview-3060--gitpod-kumquat.netlify.app/docs/introduction/languages/go)

---
<a href="https://gitpod.io/#https://github.com/gitpod-io/website/pull/3060"></a>

### CODE REVIEW

File: gitpod/docs/introduction/languages/go.md [hunk 1]
```diff
@@ -9,99 +9,70 @@ title: Go in Gitpod
 
 # Go in Gitpod
 
-Gitpod supports Go right out of the box, but there are still ways to optimize your Go experience within Gitpod.
+Gitpod supports Go right out of the box, and This guide walks you through how to fully configure a Go application using Gitpod.
```
Comment (burningion) [inline gitpod/docs/introduction/languages/go.md:12]: Gitpod includes Go in the default image, but if you need to customize your Go version or IDE setup in Gitpod, this guide will help you.

File: gitpod/docs/introduction/languages/go.md [hunk 2]
```diff
@@ -9,99 +9,70 @@ title: Go in Gitpod
...
-## Example Repositories
+## Prerequisites
 
-Here are a few Go example projects that are already automated with Gitpod:
+This guide assumes familiarity with:
 
-<div class="overflow-x-auto">
+[Docker](https://docs.docker.com/), [YAML](https://yaml.org/spec/1.1/), [Linux](https://www.linux.org/), [Bash](https://www.gnu.org/savannah-checkouts/gnu/bash/manual/bash.html) and Linux [environment variables](https://wiki.archlinux.org/title/environment_variables).
 
-| Repository                                             | Description                                               |                                                                                                                        Try It |
```
Comment (burningion) [inline gitpod/docs/introduction/languages/go.md:20]: Let's keep these example repos, they show how more advanced Go configuration works.

File: gitpod/docs/introduction/languages/go.md [hunk 3]
```diff
@@ -9,99 +9,70 @@ title: Go in Gitpod
...
-| [prometheus](https://github.com/prometheus/prometheus) | The Prometheus monitoring system and time series database | [![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/#https://github.com/prometheus/prometheus) |
-| [go-swagger](https://github.com/go-swagger/go-swagger) | A simple yet powerful representation of your RESTful API  | [![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/#https://github.com/go-swagger/go-swagger) |
-| [go-gin-app](https://github.com/gitpod-io/go-gin-app)  | Gin example running in Gitpod                             |  [![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/#https://github.com/gitpod-io/go-gin-app) |
-| [gosh-terminal](https://github.com/gosh-terminal/gosh) | A terminal implemented in Go where you can do anything    |    [![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/#https://github.com/gosh-terminal/gosh) |
+## Getting started / Quick Start
 
-</div>
+`youtube: w65POyu3ZUQ`
 
-## Workspace Configuration
```
Comment (burningion) [inline gitpod/docs/introduction/languages/go.md:29]: Why are we showing a Node/Typescript application in the Go language page?

File: gitpod/docs/introduction/languages/go.md [hunk 4]
```diff
@@ -115,16 +86,58 @@ RUN curl -fsSL https://dl.google.com/go/go${GO_VERSION}.linux-amd64.tar.gz | tar
...
+go version
+```
+
+> **Note:** If your changes are not taking effect, ensure your workspace is building from the correct [context](/docs/introduction/learn-gitpod/context-url), where your `gitpod.yml` or `gitpod.Dockerfile` are checked in to version control and are on the branch or commit that you are opening.
+
+See [configure Docker](/docs/configure/workspaces/workspace-image) for more.
+
+### Using the `dep` dependency manager in Gitpod
+
+If your project uses the [`dep` _(deprecated)_](https://golang.github.io/dep/) dependency manager then you need to add a [.gitpod.Dockerfile](/docs/configure/workspaces/workspace-image) to your project. A basic example that extends the default workspace image might be something like:
```
Comment (burningion) [inline gitpod/docs/introduction/languages/go.md:111]: if we're going to say something is deprecated, say what version number it was deprecated in the text. it saves every person having to click and look themselves.

**Review by burningion — CHANGES_REQUESTED** [review]
Added some feedback inline.

File: gitpod/docs/introduction/languages/go.md (hunk 3 above)
Comment (Siddhant-K-code) [reply gitpod/docs/introduction/languages/go.md:29]: I would fix out the wordings 👍🏽 

Regarding Video Content - We don't have any video content other than JavaScript Applications (in demo).

cc: @Nancy-Chauhan

**Review by burningion — APPROVED** [review]

**Pull request merged**